src/reducers/reviewsReducer.ts

```diff
@@ -0,0 +1,107 @@
+import { Query } from '../services/reviewsService';
+import { ReviewStats } from '../typings/reviewStats';
+import { ReviewsComments, Meta } from '../typings/reviewsComments';
+import { createReducer } from '@reduxjs/toolkit';
+import {
+  receiveReviewStats,
+  fetchReviewStatsError,
+  receiveReviewComments,
+  fetchReviewCommentsError,
+  receiveFilterAndSortReviewComments,
+  filterAndSortReviewCommentsError,
+  requestFilterAndSortReviewComments
+} from '../actions/reviewsActions';
+
+export const STATS_MAPPING = {
+  location: 'Location',
+  service: 'Service',
+  priceQuality: 'Price Quality',
+  food: 'Food',
+  room: 'Room',
+  childFriendly: 'Child Friendly',
+  interior: 'Interior',
+  size: 'Size',
+  activities: 'Activities',
+  restaurants: 'Restaurants',
+  sanitaryState: 'Sanitary State',
+  accessibility: 'Accessibility',
+  nightlife: 'Nightlife',
+  culture: 'Culture',
+  surrounding: 'Surrounding',
+  atmosphere: 'Atmosphere',
+  noviceSkiArea: 'Novice Ski Area',
+  advancedSkiArea: 'Advanced Ski Area',
+  apresSki: 'ApresSki',
+  beach: 'Beach',
+  entertainment: 'Entertainment',
+  environmental: 'Environmental',
+  pool: 'Pool',
+  terrace: 'Terrace',
+};
+
+export type initialStateType = {
+  reviewStats: ReviewStats | null,
+  reviewComments: ReviewsComments,
+  reviewCurrentQueryParams: Query | null;
+  statsLoaded: boolean,
+  statsLoadError: boolean,
+  commentsLoaded: boolean,
+  commentsLoadError: boolean,
+  currentPaginationPage: number,
+}
+
+const initialState: initialStateType = {
+  reviewStats: null,
+  reviewComments: {
+    resources: [],
+    meta: {} as Meta,
+  },
+  reviewCurrentQueryParams: null,
+  statsLoaded: false,
+  statsLoadError: false,
+  commentsLoaded: false,
+  commentsLoadError: false,
+  currentPaginationPage: 1
+};
+
+export default createReducer(initialState, {
+  [receiveReviewStats.type]: (state, action) => {
+    const { payload } = action;
+    state.reviewStats = payload;
```

Comment:
it looks like a mutation of state instead of creating a new one. Some magic from reduxjs ?

Reply:
yep, createReducer uses immer to let you write reducers as if they were mutating the state directly. In reality, the reducer receives a proxy state that translates all mutations into equivalent copy operations.